packages/openapi-generator/src/generator.ts

```diff
@@ -141,3 +110,96 @@ async function generateOpenApiService(
 function parseServiceName(filePath: string): string {
   return parse(filePath).name.replace(/-openapi$/, '');
 }
+
+/**
+ * Checks if a service already exists in the output directory.
+ * @param serviceName The name of the service to be searched.
+ * @param outputDir Path to the output directory.
+ * @returns 'true' if a duplicate service is already in the output directory.
+ */
+async function duplicateServiceExists(
+  serviceName: string,
+  outputDir: string
+): Promise<boolean> {
+  const currentOutput: string[] = await readdir(outputDir);
+  return currentOutput.some(fileName => {
+    if (serviceName === fileName) {
+      return true;
+    }
+  });
```

Comment:
I think it is not a good idea to read the directory that we are currently writing to and derive names from this. In the worst case for many files that we write asynchronously, this might even lead to indeterministic behavior. 
Instead, we should use the uniquenamefinder, that we are using for the OData generator. I moved it to util, so it becomes available for both generators in this [PR](https://github.com/SAP/cloud-sdk-js/pull/906).

However, here a little idea for improvement for this kind of code:
```suggestion
  const currentOutput = await readdir(outputDir);
  return currentOutput.some(fileName => serviceName === fileName);
```